codegen/smithy-python-codegen/src/main/java/software/amazon/smithy/python/codegen/ClientGenerator.java

```diff
@@ -334,6 +334,8 @@ async def _handle_execution(
                                 except SmithyRetryException:
                                     raise context_with_response.response
                                 await sleep(retry_token.retry_delay)
+                                if (seek := getattr(context_with_transport_request.transport_request.body, "seek")) is not None:
```

Comment:
We need to specify the default for the attribute not existing.
```suggestion
                                current_body =  context_with_transport_request.transport_request.body
                                if (seek := getattr(current_body, "seek", None)) is not None:
```

Reply:
We also probably want to implement a `seek` wrapper on `AsyncBytesReader` _or_ change our tests to use `SeekableAsyncBytesReader` too.

Reply:
`SeekableAsyncBytesReader` is a wrapper already, but applying it universally won't work for non-seekable streaming blobs that are really big. We'd have to buffer it all in memory. As for tests, I'm not sure how using it there helps. It's used in two places there:

* To wrap the actual request body and immediately read it all as `bytes`.
* To wrap the actual response body and immediately read it all as `bytes`.

The actual client will still be using the seekable one. It's also worth noting that we don't do retries in the protocol tests.